examples/src/main/java/ai/djl/examples/training/transferlearning/TrainAmazonReviewRanking.java

```diff
@@ -0,0 +1,217 @@
+    }
+
+    private static DefaultTrainingConfig setupTrainingConfig(
+            Arguments arguments, Predictor<NDList, NDList> predictor) {
+        String outputDir = arguments.getOutputDir();
+        CheckpointsTrainingListener listener = new CheckpointsTrainingListener(outputDir);
+        listener.setSaveModelCallback(
+                trainer -> {
+                    TrainingResult result = trainer.getTrainingResult();
+                    Model model = trainer.getModel();
+                    float accuracy = result.getValidateEvaluation("Accuracy");
+                    model.setProperty("Accuracy", String.format("%.5f", accuracy));
+                    model.setProperty("Loss", String.format("%.5f", result.getValidateLoss()));
+                });
+        return new DefaultTrainingConfig(Loss.softmaxCrossEntropyLoss())
+                .addEvaluator(new Accuracy())
+                .optDevices(Device.getDevices(1))
+                .optDataManager(new EmbeddingDataManager(predictor))
+                .addTrainingListeners(TrainingListener.Defaults.logging(outputDir))
+                .addTrainingListeners(listener);
+    }
+
+    private static final class BertFeaturizer implements CsvDataset.Featurizer {
+
+        private final BertFullTokenizer tokenizer;
+        private final int maxLength;
+
+        public BertFeaturizer(BertFullTokenizer tokenizer, int maxLength) {
+            this.tokenizer = tokenizer;
+            this.maxLength = maxLength;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public void featurize(DynamicBuffer buf, String input) {
+            SimpleVocabulary vocab = tokenizer.getVocabulary();
+            List<String> tokens = tokenizer.tokenize(input);
+            tokens = tokens.size() > maxLength ? tokens.subList(0, maxLength) : tokens;
+            tokens.add(0, "[CLS]");
+            tokens.add("[SEP]");
+            buf.put(vocab.getIndex("[CLS]"));
+            tokens.forEach(token -> buf.put(vocab.getIndex(token)));
+            buf.put(vocab.getIndex("[SEP]"));
+        }
+    }
+
+    private static final class EmbeddingDataManager extends DataManager {
+
+        private final Predictor<NDList, NDList> embedding;
+
+        public EmbeddingDataManager(Predictor<NDList, NDList> embedding) {
+            this.embedding = embedding;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public NDList getData(Batch batch) {
+            try {
+                // batch shape (batchsize, maximum length)
+                NDArray data = batch.getData().head();
+                long batchSize = data.getShape().get(0);
+                float maxLength = data.getShape().get(1);
+                return embedding.predict(
+                        new NDList(data, data.getManager().full(new Shape(batchSize), maxLength)));
+            } catch (TranslateException e) {
+                throw new IllegalArgumentException(e.getMessage(), e);
```

Comment:
e.getMessage() is redundent, give a better message for searching the code for debuging